src/ansys/fluent/interface/post_objects/post_objects_container.py

```diff
@@ -0,0 +1,248 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Module providing visualization objects to facilitate integration with libraries like
+Matplotlib and pyvista."""
+
+import inspect
+
+from ansys.fluent.interface.post_objects.meta import PyLocalContainer
+
+
+class Container:
+    """Base class for containers, for example, Plots, Graphics.
+
+    Parameters
+        ----------
+        session : object
+            Session object.
+        container_type: object
+            Container type (for example, Plots, Graphics)
+        module: object
+            Python module containing post definitions
+        post_api_helper: object
+            Provides helper APIs for post-processing
+        local_surfaces_provider : object, optional
+            Object providing local surfaces so that user can access surfaces
+            created in other modules, such as PyVista. The default is ``None``.
+    """
+
+    def __init__(
+        self,
+        session,
+        container_type,
+        module,
+        post_api_helper,
+        local_surfaces_provider=None,
+    ):
+        """__init__ method of Container class."""
+        session_state = container_type._sessions_state.get(session)
+        self._path = container_type.__name__
+        if not session_state:
+            session_state = self.__dict__
+            container_type._sessions_state[session] = session_state
+            self.session = session
+            self._init_module(self, module, post_api_helper)
+        else:
+            self.__dict__ = session_state
+        self._local_surfaces_provider = lambda: local_surfaces_provider or getattr(
+            self, "Surfaces", []
+        )
+
+    def get_path(self):
+        """Get container path."""
+        return self._path
+
+    @property
+    def type(self):
+        """Type."""
+        return "object"
+
+    def update(self, value):
+        """Update the value."""
+        for name, val in value.items():
+            o = getattr(self, name)
+            o.update(val)
+
+    def __call__(self, show_attributes=False):
+        state = {}
+        for name, cls in self.__dict__.items():
+            o = getattr(self, name)
+            if o is None or name.startswith("_") or name.startswith("__"):
+                continue
+
+            if cls.__class__.__name__ == "PyLocalContainer":
+                container = o
+                if getattr(container, "is_active", True):
+                    state[name] = {}
+                    for child_name in container:
```

Comment:
@prmukherj Not sure what is the type of `container` here but can we only iterate over its keys?